pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java

```diff
@@ -96,6 +96,18 @@ public ResultSetGroup execute(Request request)
     return execute(null, request);
   }
 
+  /**
+   * Executes a Pinot Request and returns the Response.
+   *
+   * @param request The request to execute
+   * @return The BrokerResponse for the request.
+   * @throws PinotClientException If an exception occurs while processing the query
+   */
+  public BrokerResponse executeRequest(Request request)
```

Comment:
Don't rename the methods as it would not be backward compatible and will break other connectors such as JDBC.

Reply:
This isn't a rename. It's a new method addition.

Reply:
My bad. @kishoreg should we have two methods both executing query - one returning the complete response while other returning the response table. IMO, we should have only one but if we remove the old one the JDBC connector will break.

Reply:
why not add the response stats as part of the existing return type. It maintains backwards compatibility and simply adds an additional method in ResultSetGroup

Reply:
I don't think the stats belong to ResultSetGroup, hence I didn't go that route. Yes, it's simple and easy method but isn't intuitive API, at least to me.

Reply:
Not really. There is scope for metadata and stats. https://cloud.google.com/spanner/docs/reference/rest/v1/ResultSet

Reply:
Yes. JDBC also follows a similar theme where you can return a ResultSetMetadata along with the actual result.

Reply:
Fixed this now by exposing stats in the ResultSetGroup itself. Please check.